### PR TITLE
Moving all proxy tests to use pod utils

### PR DESCRIPTION
**Pull request by utka**

### CODE REVIEW

**Review by fejta — APPROVED** [review]
/hold

File: prow/cluster/jobs/istio/proxy/istio.proxy.master.yaml [hunk 1]
```diff
@@ -22,13 +14,6 @@ bazel_postsubmit_spec: &bazel_postsubmit_spec
 bazel_spec: &bazel_spec
```
Comment (cjwagner) [inline prow/cluster/jobs/istio/proxy/istio.proxy.master.yaml]: This looks identical to the `bazel_postsubmit_spec`.

File: prow/cluster/jobs/istio/proxy/istio.proxy.master.yaml [hunk 2]
```diff
@@ -60,38 +45,48 @@ presubmits:
       path_alias: istio.io/istio
     labels:
       preset-service-account: "true"
-    spec: 
+    spec: &bazel_spec
```
Comment (cjwagner) [inline prow/cluster/jobs/istio/proxy/istio.proxy.master.yaml]: I'm not very familiar with YAML anchors. It looks like this assumes the `containers` field defined here and the one defined in the anchor will be merged, but I'm not sure how the parser could know if the list entries of `containers` should be merged or appended. Is the behavior in this case well defined?

IMO YAML anchors are more trouble than they're worth in most cases. Copy and pasting is super easy and super clear and I don't think redundancy is really a problem that needs to be addressed for a config file as simple as this.

File: prow/cluster/jobs/istio/proxy/istio.proxy.master.yaml [hunk 3]
```diff
@@ -101,7 +96,17 @@ postsubmits:
     context: prow/proxy-postsubmit.sh
     branches: *branch_spec
     always_run: true
+    decorate: true
+    path_alias: istio.io/proxy
+    timeout: 60m
```
Comment (cjwagner) [inline prow/cluster/jobs/istio/proxy/istio.proxy.master.yaml]: Postsubmits previously had a `120m` timeout. Was this change intentional?

**Comment by fejta** [comment]
To make this WIP either a) change the status of the PR in github to draft or b) edit the subject to start with WIP

**Review by cjwagner — COMMENTED** [review]
Whoops, I failed to click "submit review". These comments may be outdated.

File: prow/cluster/jobs/istio/proxy/istio.proxy.master.yaml (hunk 2 above)
Comment (utka) [reply prow/cluster/jobs/istio/proxy/istio.proxy.master.yaml]: yup, i was trying to figure out how to make it work without copy paste. not successfully.

File: prow/cluster/jobs/istio/proxy/istio.proxy.master.yaml (hunk 3 above)
Comment (utka) [reply prow/cluster/jobs/istio/proxy/istio.proxy.master.yaml]: done

File: prow/cluster/jobs/istio/proxy/istio.proxy.master.yaml (hunk 1 above)
Comment (utka) [reply prow/cluster/jobs/istio/proxy/istio.proxy.master.yaml]: yes, removed

**Review by cjwagner — APPROVED** [review]

**Comment by cjwagner** [comment]
This LGTM. Remove the hold and WIP when ready.
